en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/extensions/custom-filters.md

```diff
@@ -0,0 +1,112 @@
+       <artifactId>org.wso2.choreo.connect.enforcer.commons</artifactId>
+       <version>1.0.0</version>
+   </dependency>
+   ```
+
+2. Use the following interface to implement custom filter.
+
+    ```java
+    package org.wso2.choreo.connect.enforcer.commons;
+    
+    import org.wso2.choreo.connect.enforcer.commons.model.APIConfig;
+    import org.wso2.choreo.connect.enforcer.commons.model.RequestContext;
+    
+    import java.util.Map;
+    
+    /**
+        * Filters are the request interceptors that applies API Management capabilities at the gateway layer. This is the
+        * Interface to implement chain of filters at the gateway.
+        */
+    public interface Filter {
+    
+        default void init(APIConfig apiConfig, Map<String, String> configProperties){};
+    
+        boolean handleRequest(RequestContext requestContext);
+    }
+    ```
+
+    Here is the sample filter implementation which reads the property called `CustomProperty` from Filter 
+    Configuration and add it as a header to the request.
+
+    ```java
+    package org.example.tests;
+    
+    import org.wso2.choreo.connect.enforcer.commons.model.APIConfig;
+    import org.wso2.choreo.connect.enforcer.commons.model.RequestContext;
+    import org.wso2.choreo.connect.enforcer.commons.Filter;
+    
+    import java.util.Map;
+    
+    public class CustomFilter implements Filter {
+        private static final Logger log = LoggerFactory.getLogger(CustomFilter.class);
+        private Map<String, String> configProperties;
+    
+        @Override
+        public void init(APIConfig apiConfig, Map<String, String> configProperties) {
+            this.configProperties = configProperties;
+        }
+    
+        @Override
+        public boolean handleRequest(RequestContext requestContext) {
+            String headerValue = configProperties.get("CustomProperty");
+            requestContext.addOrModifyHeaders("Custom-header-1", headerValue);
+            return true;
+        }
+    }
+    ```
+
+3. Since we use JAVA SPI (Service Provider Interface), We need to provide the provider configuration file 
+`META-INF/services/org.wso2.choreo.connect.enforcer.commons.Filter`. If you are using maven, create the file this
+file inside the <Project>/src/main/resources directory. The content of the file needs to be the qualified class name
+of the filter implementation.
+
+    ```
+    org.example.tests.CustomFilter
+    ```
+
+4. Build the project and create the jar.
+
+5. Edit the enforcer related configuration within config.toml file to include the custom filter. The classname needs to
+be the fully qualified className. The position denotes the final filter position in the chain, when all the filters 
+are added. By default, the first position is taken by Authentication Filter and the Thorttle Filter is placed as the 
+second filter. As the below example configuration contains `1` as the Position, it would be executed prior to 
```

Comment:
```suggestion
second filter. As the below example configuration contains `1` as the `position`, it would be executed prior to 
```
The position value is 3 in the sample provided, not 1. Am I missing something here?